manifests/puppetserver.pp

```diff
@@ -16,6 +16,33 @@
     metric_ensure => $metrics_ensure,
     hosts         => $hosts,
     metrics_port  => $port,
-  }
+    additional_metrics => $additional_metrics,
 
-}
+  }
+  
+  if versioncmp($facts['pe_server_version'], '2018.1.0') < 0 {
```

Comment:
Variables must be declared before they are used. This `if` block has to appear before the `puppet_metrics_collector::pe_metric` that uses it.